### PR TITLE
DOC-308: Documentation for conf file

### PR DESCRIPTION
**Pull request by rahav-certora**
Links to generated documentation:
 - [Conf file API](https://certora-certora-prover-documentation--134.com.readthedocs.build/en/134/docs/prover/cli/conf-file-api.html)
 - [Options](https://certora-certora-prover-documentation--134.com.readthedocs.build/en/134/docs/prover/cli/options.html)

### CODE REVIEW

File: docs/prover/cli/conf-file-api.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+**String Value CLI Options**
+
+Flags in CLI API that accept a single string will be stored as **JSON Strings**. Example:
+```
+"solc": "solc4.25"
+```
+**Number Value CLI Options**
+
+Flags in CLI API that accept numbers will be stored as **JSON Strings** not as **JSON Numbers**. Example:
```
Comment (chandrakananandi) [inline docs/prover/cli/conf-file-api.md:35]: curious why this is the case. This is not related to this PR ofcourse.

File: docs/prover/cli/conf-file-api.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+```
+    "solc_map": {
+        "A": "solc5.11",
+        "B": "solc5.9",
+        "C": "solc6.8"
+    }
+    
+```
+
+<div style="background-color: #FFFFE0; padding: 10px; border: 1px solid #E6DB55;">
```
Comment (chandrakananandi) [inline docs/prover/cli/conf-file-api.md:70]: i like these yellow boxes! I do wonder if these should be
```
```{note}
content
```

```

instead? That seems to be what we do elsewhere in the doc when we have similar content.

File: docs/prover/cli/conf-file-api.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+(to be more precise the format is [JSON5](https://json5.org/)):
+
+```bash
+certoraRun my_params.conf
+```
+<div style="background-color: #FFFFE0; padding: 10px; border: 1px solid #E6DB55;">
+To denote a text file as a conf file use the <strong>.conf</strong> suffix
+</div>
+
+Converting from CLI Options to JSON
```
Comment (chandrakananandi) [inline docs/prover/cli/conf-file-api.md:16]: This section should say how to automatically get the conf from a runscript.

**Review by chandrakananandi — CHANGES_REQUESTED** [review]
Thanks for putting this together! I think it would be good to make it more complete. Maybe a table mapping the CLI flag names to their corresponding json field names? I recall some were not obvious.

File: docs/prover/cli/conf-file-api.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        "SolcArgs/C.sol"
+    ],
+    "multi_assert_check": true,
+    "solc_map": {
+        "B": "solc6.1",
+        "C": "solc5.12",
+        "SolcArgs/A.sol": "solc6.1"
+    },
+    "verify": "A:SolcArgs/Trigger.spec"
+}
```
[the remark this answers is not in the document]
Comment (chandrakananandi) [reply docs/prover/cli/conf-file-api.md:99]: I also recall there used to be the env vars: ` CERTORA_NEW_API` and `CERTORA_OLD_API ` to switch between the old and new json format? Is that still supported?

[the remark this answers is not in the document]
Comment (chandrakananandi) [reply docs/prover/cli/conf-file-api.md:99]: if so, maybe we should mention it?

File: docs/prover/cli/conf-file-api.md (hunk 2 above)
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:70]: My intention was not to mark it as a note, more for emphasizing the text without breaking the flow. I removed the yellow boxes.

File: docs/prover/cli/conf-file-api.md (hunk 3 above)
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:16]: I renamed the title to "How CLI Options are mapped to JSON". I later described where to find the generated conf file

File: docs/prover/cli/conf-file-api.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:99]: In the new API I removed all the inconsistencies between flags and conf keys. Now there are no exceptions!
This documentation if for the new api only. We announced that the old API is not going to be supported after the end of August. CERTORA_NEW_API/CERTORA_OLD_API will soon be removed

File: docs/prover/cli/conf-file-api.md (hunk 2 above)
Comment (chandrakananandi) [reply docs/prover/cli/conf-file-api.md]: i liked them :)

File: docs/prover/cli/conf-file-api.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (chandrakananandi) [reply docs/prover/cli/conf-file-api.md:101]: got it!

**Review by chandrakananandi — APPROVED** [review]
Thanks for this PR!

File: docs/prover/cli/conf-file-api.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+certoraRun my_params.conf
+```
+
+Conf files must use the <strong>.conf</strong> suffix
+
+
+How CLI Options are mapped to JSON
+----------------------------------
+
+**JSON Keys**
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:19]: Use the heading commands (e.g. `### JSON Keys`) rather than bolding to create headers.

Comment (mdgeorge4153) [reply docs/prover/cli/conf-file-api.md:19]: Although in this case, I would probably skip the headers in favor of a bulleted list.  E.g, something like
> Command-line arguments are stored as key-value pairs in the conf file.  The keys are the names of the parameters (with the leading `--` removed).  For example,
> ```{sh}
> certoraRun --verify Example:example.spec
> ```
> is equivalent to running with the following conf file:
> ```{json}
> { "verify": "Example:example.spec" }
> ```
> 
> The values in the map depend on the type of arguments:
>  - for flags that take no additional arguments (such as {ref}`--send_only`), the value should be `true`.  For example,
>    ```{sh}
>    certoraRun --send_only
>    ```
>    would be encoded as:
>    ```{json}
>    { "send_only": true }
>    ```
>  
>  - flags that take a single additional argument (such as {ref}`--solc` or {ref}`--smt_timeout`) are encoded as a JSON string.  For example,
>    ```{sh}
>    certoraRun --solc solc4.25 --smt_timeout 600
>    ```
>    would be encoded as:
>    ```{json}
>    { "solc": "solc4.25", "smt_timeout": "600" }
>    ```
>    Note that conf files do not use JSON numbers; numbers are encoded as strings.
>  
>   - flags that take multiple additional arguments (such as {ref}`--packages`) are encoded as JSON lists.  For example,
>     ```{sh}
>     certoraRun --packages @balancer-labs/v2-solidity-utils=pkg/solidity-utils @balancer-labs/v2-vault=pkg/vault"
>     ```
>     would be encoded as:
>     ```{json}
>     {
>         "packages": [
>             "@balancer-labs/v2-solidity-utils=pkg/solidity-utils",
>             "@balancer-labs/v2-vault=pkg/vault"
>         ]
>     }
>     ```

File: docs/prover/cli/conf-file-api.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    }
+    
+```
+
+A conf file is generated each time `certoraRun` completes execution successfully.
+The conf file is stored as `**run.conf**` in the build directory under **`.certora_internal`**.
+<p>Conf file of the latest run can be found in:
+
+```
+./.certora_internal/latest/run.conf
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:75]: I would omit the `./`, and would probably just inline this, since it is not a command

File: docs/prover/cli/conf-file-api.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+The **Conf File API** is an alternative API for  `certoraRun` . In terms of functionality 
+this API is identical to the [CLI Options](options.md) API. Instead of calling `certoraRun` 
+with a list of shell flags, some or all the flags can be stored in a [JSON](https://www.json.org/json-en.html) file 
+(to be more precise the format is [JSON5](https://json5.org/)):
+
+```
+certoraRun my_params.conf
+```
+
+Conf files must use the <strong>.conf</strong> suffix
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:13]:
```suggestion
Conf files must use the `.conf` suffix
```

File: docs/prover/cli/conf-file-api.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
+Certora Prover Conf File API
+============================
+
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:3]: We should probably mention that the VSCode extension, and also the fact that by convention conf files are stored in `certora/conf` (I think that's right?)

**Review by mdgeorge4153 — CHANGES_REQUESTED** [review]
Looks good.  I left several comments; let me know if you want to make an editing pass.

File: docs/prover/cli/conf-file-api.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
+Certora Prover Conf File API
```
[the remark this answers is not in the document]
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:1]: done

File: docs/prover/cli/conf-file-api.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+============================
+
+The **Conf File API** is an alternative API for  `certoraRun` . In terms of functionality 
+this API is identical to the [CLI Options](options.md) API. Instead of calling `certoraRun` 
+with a list of shell flags, some or all the flags can be stored in a [JSON](https://www.json.org/json-en.html) file 
+(to be more precise the format is [JSON5](https://json5.org/)):
```
[the remark this answers is not in the document]
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:7]: done

File: docs/prover/cli/conf-file-api.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+```
+    "solc_map": {
+        "A": "solc5.11",
+        "B": "solc5.9",
+        "C": "solc6.8"
+    }
+    
+```
+
+A conf file is generated each time `certoraRun` completes execution successfully.
```
[the remark this answers is not in the document]
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:70]: done

File: docs/prover/cli/conf-file-api.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+The conf file is stored as `**run.conf**` in the build directory under **`.certora_internal`**.
```
[the remark this answers is not in the document]
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:71]: done

File: docs/prover/cli/conf-file-api.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+<p>Conf file of the latest run can be found in:
```
[the remark this answers is not in the document]
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:72]: done

File: docs/prover/cli/conf-file-api.md (hunk 6 above)
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:75]: done

File: docs/prover/cli/conf-file-api.md (hunk 8 above)
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:3]: Added a section for VS code users

**Review by shellygr — APPROVED** [review]

File: docs/prover/cli/conf-file-api.md [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+```
+certoraRun --send_only
+```
+
+would be encoded as:
+```
+{ "send_only": true }
+```
+
+* flags that expect a single argument (such as {ref}`--solc`) or as {ref}`--loop_iter`)
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:45]: Capitalize start of sentence

File: docs/prover/cli/conf-file-api.md [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+```
+is equivalent to running with the following conf file:
+
+```
+{ "verify": "Example:example.spec" }
+```
+The values in the map depend on the type of arguments:
+
+* boolean flags that take no arguments (such as {ref}`--send_only`), 
+the value should be `true`. For example,
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:35]: Incomplete sentence.  Perhaps "Boolean flags that take no arguments (such as `--send_only`) should have a value of `true`."

File: docs/prover/cli/conf-file-api.md [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+certoraRun --verify Example:example.spec
+```
+is equivalent to running with the following conf file:
+
+```
+{ "verify": "Example:example.spec" }
+```
+The values in the map depend on the type of arguments:
+
+* boolean flags that take no arguments (such as {ref}`--send_only`),
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:34]: capitalize start of sentence

File: docs/prover/cli/conf-file-api.md [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+is equivalent to running with the following conf file:
+
+```
+{ "verify": "Example:example.spec" }
+```
+The values in the map depend on the type of arguments:
+
+* boolean flags that take no arguments (such as {ref}`--send_only`), 
+the value should be `true`. For example,
+```
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:36]:
````suggestion
```sh
````

File: docs/prover/cli/conf-file-api.md [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+{
+"packages": [
+    "@balancer-labs/v2-solidity-utils=pkg/solidity-utils",
+    "@balancer-labs/v2-vault=pkg/vault"
+    ]
+}
+```
+
+
+* The input files in the CLI API will be stored under the key **files**
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:74]:
```suggestion
* The input files in the CLI API will be stored under the key `files`
```

File: docs/prover/cli/conf-file-api.md [hunk 19]
```diff
@@ -0,0 +1,143 @@
...
+{
+    ...
+    "files": "example.sol" 
+    ...
+}
+```
+
+**Map Value CLI Options**
+
+Flags in CLI API that are maps will be stored as **JSON Objects**. Example:
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:90]: How do I know whether a flag is a map?

File: docs/prover/cli/conf-file-api.md [hunk 20]
```diff
@@ -0,0 +1,143 @@
...
+```
+    "solc_map": {
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:92]: It would be good to have the command line as you do for the above examples.

File: docs/prover/cli/conf-file-api.md [hunk 21]
```diff
@@ -0,0 +1,143 @@
...
+        "A": "solc5.11",
+        "B": "solc5.9",
+        "C": "solc6.8"
+    }
+    
+```
+**Generating a Conf File**
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:99]: Use sectioning commands instead of bolding

File: docs/prover/cli/conf-file-api.md [hunk 22]
```diff
@@ -0,0 +1,143 @@
...
+
+After each successful run of `certoraRun` a conf file is generated and is
+stored in the file **run.conf** under the internal directory of that run.
+The conf file of the latest run can be found in:
+
+```
+.certora_internal/latest/run.conf
+```
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:107]:
```suggestion
stored in the file `.certora_internal/latest/run.conf`.
```

File: docs/prover/cli/conf-file-api.md [hunk 23]
```diff
@@ -0,0 +1,143 @@
...
+
+Instead of generating a complete conf file from scratch, users can take 
+one of these generated  conf files as a basis for their modifications.
+
+**Conf Files in the VS Code IDE Extension**
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:112]: sectioning instead of bold

File: docs/prover/cli/conf-file-api.md [hunk 24]
```diff
@@ -0,0 +1,143 @@
...
+
+VS Code users can generate conf files using the [Certora IDE Extension](https://marketplace.visualstudio.com/items?itemName=Certora.vscode-certora-prover). The extension 
+offers an intuitive UI for configuring Prover jobs. Each job
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:115]: It's a little unclear what the UI is or how to access it, even though I think that's what you're describing in this paragraph.  I think it's because it seems like you're saying two different things: the VSCode extension can be used to make config files, and also the VSCode extension uses config files.

I think something like

"The [Certora IDE VS Code extension](...) automatically generates conf files for each configured job; these files are saved in the `certora/confs` directory and can also be ..."

File: docs/prover/cli/conf-file-api.md [hunk 25]
```diff
@@ -0,0 +1,143 @@
...
+offers an intuitive UI for configuring Prover jobs. Each job 
+keeps a conf file that allows rerunning the job. All conf files
+are stored in the VS code project under the folder  `certora/confs`. A link to the job's conf file
+can also be found in the files section of the run report once the job is completed.
+
+**Complete Example**
+
+The command line:
+```
+certoraRun SolcArgs/A.sol SolcArgs/A.sol:B SolcArgs/C.sol --verify A:SolcArgs/Trigger.spec --solc_map SolcArgs/A.sol=solc6.1,B=solc6.1,C=solc5.12 --multi_assert_check
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:124]: I would break this into several lines - the generated page is hard to read because you have to scroll.

File: docs/prover/cli/conf-file-api.md [hunk 26]
```diff
@@ -0,0 +1,143 @@
+Certora Prover Conf File Format
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:1]: I'd consider shortening this to "Conf files".  Also the CLI Options page, now that I'm thinking about it.

File: docs/prover/cli/options.md [hunk 27]
```diff
@@ -243,7 +243,7 @@ When we do not care much for the output. It is recommended when running the tool
```
Comment (mdgeorge4153) [inline docs/prover/cli/options.md:243]: There should probably be a note somewhere in the CLI options page that mentions that you can run with a conf file and links to the conf file page.

**Review by mdgeorge4153 — CHANGES_REQUESTED** [review]
Looking good, a few small editing issues.

File: docs/prover/cli/options.md (hunk 27 above)
Comment (rahav-certora) [reply docs/prover/cli/options.md:243]: done

File: docs/prover/cli/conf-file-api.md (hunk 26 above)
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:1]: changed in both places

File: docs/prover/cli/conf-file-api.md (hunk 25 above)
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:124]: done

File: docs/prover/cli/conf-file-api.md (hunk 24 above)
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:115]: done

File: docs/prover/cli/conf-file-api.md (hunk 23 above)
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:112]: replaced with ## 
is that the right level?

File: docs/prover/cli/conf-file-api.md (hunk 16 above)
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:34]: done

[the remark this answers is not in the document]
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:34]: fixed

File: docs/prover/cli/conf-file-api.md (hunk 15 above)
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:35]: done

File: docs/prover/cli/conf-file-api.md (hunk 14 above)
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:45]: done

File: docs/prover/cli/conf-file-api.md [hunk 28]
```diff
@@ -0,0 +1,143 @@
...
+* The input files in the CLI API will be stored under the key **files**
+
+```
+certoraRun example.sol  ...
+```
+will appear in the conf file as:
+```
+{
+    ...
+    "files": "example.sol"
```
[the remark this answers is not in the document]
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:83]: good catch, fixed

File: docs/prover/cli/conf-file-api.md [hunk 29]
```diff
@@ -0,0 +1,143 @@
...
+will appear in the conf file as:
+```
+{
+    ...
+    "files": "example.sol" 
+    ...
+}
+```
+
+**Map Value CLI Options**
```
[the remark this answers is not in the document]
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:88]: done

File: docs/prover/cli/conf-file-api.md (hunk 19 above)
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:90]: listed the 2 map flags --solc_map and --solc_optimize_map

File: docs/prover/cli/conf-file-api.md (hunk 20 above)
Comment (rahav-certora) [reply docs/prover/cli/conf-file-api.md:92]: done

File: docs/prover/cli/conf-file-api.md [hunk 30]
```diff
@@ -0,0 +1,166 @@
...
+certoraRun SolcArgs/A.sol SolcArgs/A.sol:B SolcArgs/C.sol \
+  --verify A:SolcArgs/Trigger.spec \
+  --solc_map SolcArgs/A.sol=solc6.1,B=solc6.1,C=solc5.12 \
+  --multi_assert_check 
+
+
+```
+
+will generate the conf file below:
+```
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:151]:
````suggestion
```json
````

File: docs/prover/cli/conf-file-api.md [hunk 31]
```diff
@@ -0,0 +1,166 @@
...
+.certora_internal/latest/run.conf
+```
+
+Instead of generating a complete conf file from scratch, users can take 
+one of these generated conf files as a basis for their modifications.
+
+## Conf Files in the VS Code IDE Extension
+The [Certora IDE Extension](https://marketplace.visualstudio.com/items?itemName=Certora.vscode-certora-prover)
+automatically generates conf files for each configured job; these conf files
+are stored in the VS code project under the folder  `certora/confs`.
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:134]:
```suggestion
are stored in the VS Code project under the folder  `certora/confs`.
```

File: docs/prover/cli/conf-file-api.md [hunk 32]
```diff
@@ -0,0 +1,166 @@
...
+Once the job is completed, a link to the job's conf file can also be found in the files section of the 
+run report.
+
+### Complete Example
+
+The command line:
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:140]:
```suggestion
The command line
```

File: docs/prover/cli/conf-file-api.md [hunk 33]
```diff
@@ -0,0 +1,166 @@
...
+
+```
+certoraRun my_params.conf
+```
+
+* Conf files must use the `.conf` suffix
+
+
+
+How CLI Options are mapped to JSON
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:18]:
```suggestion
How CLI options are mapped to JSON
```

File: docs/prover/cli/conf-file-api.md [hunk 34]
```diff
@@ -0,0 +1,166 @@
...
+  would be encoded as:
+    ```
+    "solc_optimize_map": {
+        "A": "200",
+        "B": "200",
+        "C": "300"
+    }
+    
+    ```
+## Generating a Conf File
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:118]:
```suggestion
## Generating a conf file
```

File: docs/prover/cli/conf-file-api.md [hunk 35]
```diff
@@ -0,0 +1,166 @@
...
+The conf file of the latest run can be found in:
+
+```
+.certora_internal/latest/run.conf
+```
+
+Instead of generating a complete conf file from scratch, users can take 
+one of these generated conf files as a basis for their modifications.
+
+## Conf Files in the VS Code IDE Extension
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:131]:
```suggestion
## Conf files in the VS Code IDE extension
```

File: docs/prover/cli/conf-file-api.md [hunk 36]
```diff
@@ -0,0 +1,166 @@
...
+The [Certora IDE Extension](https://marketplace.visualstudio.com/items?itemName=Certora.vscode-certora-prover)
+automatically generates conf files for each configured job; these conf files
+are stored in the VS code project under the folder  `certora/confs`.
+Once the job is completed, a link to the job's conf file can also be found in the files section of the 
+run report.
+
+### Complete Example
```
Comment (mdgeorge4153) [inline docs/prover/cli/conf-file-api.md:138]:
```suggestion
### Complete example
```

File: docs/prover/cli/options.md [hunk 37]
```diff
@@ -819,3 +824,9 @@ splits generated here is equal to `2^n` where `n` is the initial splitting depth
 (unless the program has less than `n` branchings, which will be rare in
 practice).
 
+Using Conf Files 
+----------------
+For larger projects, the command line for running the Certora Prover can become large
```
Comment (mdgeorge4153) [inline docs/prover/cli/options.md:829]: I would probably move this paragraph right into the intro, since it is a general recommendation.  Otherwise I worry it will get lost here.

File: docs/prover/cli/options.md [hunk 38]
```diff
@@ -819,3 +824,9 @@ splits generated here is equal to `2^n` where `n` is the initial splitting depth
 (unless the program has less than `n` branchings, which will be rare in
 practice).
 
+Using Conf Files 
+----------------
+For larger projects, the command line for running the Certora Prover can become large 
+and cumbersome. It is therefore recommended to use configuration files instead. 
+These are [JSON5](https://json5.org/) files (with .conf extension) that hold the parameters and options for the Prover.
+See more on conf file [here3](./conf-file-api.md)
```
Comment (mdgeorge4153) [inline docs/prover/cli/options.md:832]:
```suggestion
See {ref}`conf-files` for more information.
```

**Review by mdgeorge4153 — APPROVED** [review]
Looks great.  I approved but added a few small formatting tweaks; please address those before merging.

File: docs/prover/cli/options.md (hunk 37 above)
Comment (rahav-certora) [reply docs/prover/cli/options.md:829]: done
